File: lib_book/src/main/java/com/ebook/basebook/mvp/model/impl/WebBookModelImpl.java
package com.ebook.basebook.mvp.model.impl;


import com.ebook.basebook.cache.ACache;
import com.ebook.basebook.mvp.model.StationBookModel;
import com.ebook.basebook.mvp.model.WebBookModel;
import com.ebook.db.entity.BookContent;
import com.ebook.db.entity.BookShelf;
import com.ebook.db.entity.Library;
import com.ebook.db.entity.SearchBook;
import com.ebook.db.entity.WebChapter;

import java.util.List;

import io.reactivex.Observable;

public class WebBookModelImpl implements WebBookModel {
    private volatile static WebBookModel bookModel;
    private final StationBookModel stationBookModel;

    public WebBookModelImpl(StationBookModel iStationBookModel) {
        this.stationBookModel = iStationBookModel;
    }

    public static WebBookModel getInstance() {
        if (bookModel == null) {
            synchronized (WebBookModelImpl.class) {
                if (bookModel == null) {
                    //更换书源只需要修改这一行
                    bookModel = new WebBookModelImpl(BiQuGeBookModelImpl.getInstance());
                }
            }
        }
        return bookModel;
    }

    @Override
    public Observable<BookShelf> getBookInfo(BookShelf bookShelf) {
        return stationBookModel.getBookInfo(bookShelf);
    }

    @Override
    public Observable<WebChapter<BookShelf>> getChapterList(final BookShelf bookShelf) {
        return stationBookModel.getChapterList(bookShelf);
    }

    @Override
    public Observable<BookContent> getBookContent(String durChapterUrl, int durChapterIndex) {
        return stationBookModel.getBookContent(durChapterUrl, durChapterIndex);
    }

    @Override
    public Observable<List<SearchBook>> getKindBook(String url, int page) {
        return stationBookModel.getKindBook(url, page);
    }

    @Override
    public Observable<Library> getLibraryData(ACache aCache) {
        return stationBookModel.getLibraryData(aCache);
    }

    @Override
    public Observable<Library> analyzeLibraryData(String data) {
        return stationBookModel.analyzeLibraryData(data);
    }

    @Override
    public Observable<List<SearchBook>> searchBook(String content, int page) {
        return stationBookModel.searchBook(content, page);
    }
}

File: lib_book_common/src/main/java/com/ebook/common/analyze/impl/WebBookModelImpl.kt
package com.ebook.common.analyze.impl

import android.content.Context
import com.ebook.api.cache.ACache
import com.ebook.common.analyze.StationBookModel
import com.ebook.common.analyze.WebBookModel
import com.ebook.db.entity.BookContent
import com.ebook.db.entity.BookShelf
import com.ebook.db.entity.Library
import com.ebook.db.entity.SearchBook
import com.ebook.db.entity.WebChapter
import io.reactivex.rxjava3.core.Observable


object WebBookModelImpl : WebBookModel {
    private val stationBookModel: StationBookModel = TXTDownloadBookModelImpl
    override fun getBookInfo(bookShelf: BookShelf): Observable<BookShelf> {
        return stationBookModel.getBookInfo(bookShelf)
    }

    override fun getChapterList(bookShelf: BookShelf): Observable<WebChapter<BookShelf>> {
        return stationBookModel.getChapterList(bookShelf)
    }

    override fun getBookContent(
        context: Context,
        durChapterUrl: String,
        durChapterIndex: Int
    ): Observable<BookContent> {
        return stationBookModel.getBookContent(context, durChapterUrl, durChapterIndex)
    }

    override fun getKindBook(url: String, page: Int): Observable<List<SearchBook>> {
        return stationBookModel.getKindBook(url, page)
    }

    override fun getLibraryData(aCache: ACache): Observable<Library> {
        return stationBookModel.getLibraryData(aCache)
    }

    override fun analyzeLibraryData(data: String): Observable<Library> {
        return stationBookModel.analyzeLibraryData(data)
    }

    override fun searchBook(content: String, page: Int): Observable<List<SearchBook>> {
        return stationBookModel.searchBook(content, page)
    }
}
